### DISCUSSION TITLE
Ubuntu v20.04.5 Good but v22.04.1 Not?

### DISCUSSION THREAD

**Discussion by PhilipElder** [mastodon/mastodon#19651] [General]
Hi there,

We have been setting up Mastodon on Ubuntu v20.04.1-5 without issue.

We tried to set it up on v22.04.1 ending up with things not working out. The site would present a logon page with no CSS styling then a white page once logged on. We ran through it twice using the setup process from v20.04.1-5 but can't seem to get it to work.

When troubleshooting page rendering what's the best way to go about it?

Obviously, something has changed between Ubuntu v20 and v22 but I'm not 100% sure where that is yet.

Suggestions please?

Thank you! :0)

**Comment by trwnh** [comment]
OpenSSL 3, I think

**Comment by hperrin** [comment]
I'm also seeing this same thing. The CSS and JS requests are all resulting in a 404 from `server: Mastodon`.

**Reply by hperrin** [reply]
Without the proxy try_files, they result in a 403 from `server: nginx/1.18.0 (Ubuntu)`.

**Reply by hperrin** [reply]
My guess is that the `www-data` user cannot access the files in `/home/mastodon/live/`.

**Reply by hperrin** [reply]
Yep, that was it. Run `sudo usermod -a -G mastodon www-data` to add `www-data` to the `mastodon` group, then restart Nginx, and everything works.

**Reply by trwnh** [reply]
There's an issue with precompiling assets, I think because of OpenSSL incompatibilities within Webpack and specifically within the versions packaged by Ubuntu 22.04.

EDIT: I see that it was something else in this case, luckily you found the cause!

**Comment by PhilipElder** [comment]
`sudo usermod -a -G mastodon www-data `
Okay, I had to get the instance into production. 
I'll stand up a v22.04.1 LTS and test to confirm.
Thank you! :0)

**Comment by HankLloydRight** [comment]
Exact same problem here.. on 22.404.1  LTS on AWS 
I followed the install page to the letter, and got the Mastodon banner but no CSS or JS was being loaded (all 404).  Glad I found this discussion!
`sudo usermod -a -G mastodon www-data`  and restarting NGINX is all that I needed.  
`www-data` was already in the `mastodon` group.
The install page at: https://docs.joinmastodon.org/admin/install/ should be updated.
Thank you @hperrin !
